### PR TITLE
EU-66

### PR DESCRIPTION
**Pull request by victor-nuzhniy**

### CODE REVIEW

File: migrations/main.sql [hunk 1]
```diff
@@ -672,3 +672,42 @@ CREATE VIEW user_documents_list_view AS
         ur.university_id,
         ud.user_document_id,
         ur.user_id;
+
+DROP VIEW IF EXISTS user_request_hostel_warrant_view;
+CREATE VIEW user_request_hostel_warrant_view AS
+    SELECT
+    urr.user_request_review_id,
+    urr.room_number as room_number,
```
Comment (anton-serozhechkin) [inline migrations/main.sql]: not need "as ..."

File: migrations/main.sql [hunk 2]
```diff
@@ -672,3 +672,42 @@ CREATE VIEW user_documents_list_view AS
...
+    urr.user_request_id as user_request_id,
```
Comment (anton-serozhechkin) [inline migrations/main.sql]: not need "as ..."

File: migrations/main.sql [hunk 3]
```diff
@@ -672,3 +672,42 @@ CREATE VIEW user_documents_list_view AS
...
+    urr.created_at as created_at,
```
Comment (anton-serozhechkin) [inline migrations/main.sql]: not need "as ..."

File: migrations/main.sql [hunk 4]
```diff
@@ -672,3 +672,42 @@ CREATE VIEW user_documents_list_view AS
...
+    h.number as hostel_number,
+    h.street as hostel_street,
+    h.build as hostel_build,
+    bp.bed_place_name as bed_place_name,
```
Comment (anton-serozhechkin) [inline migrations/main.sql]: not need "as ..."

File: migrations/main.sql [hunk 5]
```diff
@@ -672,3 +672,42 @@ CREATE VIEW user_documents_list_view AS
...
+    u.university_name as university_name,
```
Comment (anton-serozhechkin) [inline migrations/main.sql]: not need "as ..."

File: migrations/main.sql [hunk 6]
```diff
@@ -672,3 +672,42 @@ CREATE VIEW user_documents_list_view AS
...
+    u.short_university_name as short_university_name,
+    u.city as university_city,
+    ur.status_id as status_id,
+    ur.user_id as user_id,
```
Comment (anton-serozhechkin) [inline migrations/main.sql]: not need "as ..."

File: migrations/main.sql [hunk 7]
```diff
@@ -672,3 +672,42 @@ CREATE VIEW user_documents_list_view AS
...
+    urr.user_request_id as user_request_id,
+    urr.created_at as created_at,
+    h.number as hostel_number,
+    h.street as hostel_street,
+    h.build as hostel_build,
+    bp.bed_place_name as bed_place_name,
+    u.university_name as university_name,
+    u.short_university_name as short_university_name,
+    u.city as university_city,
+    ur.status_id as status_id,
```
Comment (anton-serozhechkin) [inline migrations/main.sql]: not need "as ..."

File: migrations/main.sql [hunk 8]
```diff
@@ -672,3 +672,42 @@ CREATE VIEW user_documents_list_view AS
...
+    urr.user_request_review_id,
+    urr.room_number as room_number,
+    urr.user_request_id as user_request_id,
+    urr.created_at as created_at,
+    h.number as hostel_number,
+    h.street as hostel_street,
+    h.build as hostel_build,
+    bp.bed_place_name as bed_place_name,
+    u.university_name as university_name,
+    u.short_university_name as short_university_name,
```
Comment (anton-serozhechkin) [inline migrations/main.sql]: not need "as ..."

File: apps/services/handlers.py [hunk 9]
```diff
@@ -388,6 +391,46 @@ async def __generate_user_document_name_for_download(
             f"{student.last_name}.docx"
         )
 
+    @classmethod
+    async def __create_user_warrant(
```
Comment (anton-serozhechkin) [inline apps/services/handlers.py]: let's move this method under method download_warrant_document

File: apps/services/handlers.py [hunk 10]
```diff
@@ -388,6 +391,46 @@ async def __generate_user_document_name_for_download(
...
+            data={
+                "name": "Ордер на поселення в гуртожиток",
+                "content": content,
+                "user_request_id": data.user_request_id,
+            },
+        )
+        return user_document_record
+
+    @classmethod
+    async def __create_user_warrant_content(
```
Comment (anton-serozhechkin) [inline apps/services/handlers.py]: let's move this method under method download_warrant_document

File: apps/services/handlers.py [hunk 11]
```diff
@@ -424,5 +467,31 @@ async def create_students_list_from_file(
...
+        *,
+        request: Request,
+        user_request_review_id: int,
+        session: AsyncSession,
+    ) -> Tuple[str, str]:
+        warrant_view_data = await user_request_hostel_warrant_view_service.read(
+            session=session, data={"user_request_review_id": user_request_review_id}
+        )
+        check_user_request_status(warrant_view_data.status_id)
+        user_document_list = await user_document_service.list(
```
Comment (anton-serozhechkin) [inline apps/services/handlers.py]: i'm not sure that we need this

File: apps/services/handlers.py [hunk 12]
```diff
@@ -424,5 +467,31 @@ async def create_students_list_from_file(
             students.append(student)
         return await student_service.create_many(session=session, objs=students)
 
+    async def download_warrant_document(
+        self,
```
Comment (anton-serozhechkin) [inline apps/services/handlers.py]: as i remember, u started to create static methods instead of self methods

File: apps/services/handlers.py [hunk 13]
```diff
@@ -388,6 +391,46 @@ async def __generate_user_document_name_for_download(
             f"{student.last_name}.docx"
         )
 
+    @classmethod
+    async def __create_user_warrant(
+        cls, session: AsyncSession, data
+    ) -> UserDocument:
+        content = await cls.__create_user_warrant_content(data)
+        user_document_record = await user_document_service.create(
```
Comment (anton-serozhechkin) [inline apps/services/handlers.py]: we don't need to create a new record, just generate file and return it - that's all

**Review by anton-serozhechkin — CHANGES_REQUESTED** [review]

**Comment by anton-serozhechkin** [comment]
@victor-nuzhniy also u forgot to use the value of column "hostel_build" in .docx template

**Review by anton-serozhechkin — DISMISSED** [review]

File: apps/services/utils.py [hunk 14]
```diff
@@ -119,3 +119,12 @@ def check_file_existing(path: str) -> None:
             message=f"File with path {path} was removed or deleted",
             code=http_status.HTTP_409_CONFLICT,
         )
+
+
+def check_user_request_status(status_id: int) -> None:
+    if status_id != 1:
```
Comment (Kostiantyn-Salnykov) [inline apps/services/utils.py]: What does it mean? Maybe it's better to add some Enum, that will hold a `1` value, instead of a hardcoded value.

**Comment by victor-nuzhniy** [comment]
Error appears when checking download-warrant-document working

**Comment by victor-nuzhniy** [comment]
Everything all right

File: apps/services/utils.py (hunk 14 above)
Comment (anton-serozhechkin) [reply apps/services/utils.py]: fixed

**Review by anton-serozhechkin — APPROVED** [review]

**Pull request merged**